src/server/TCPServer.cpp

```diff
@@ -110,6 +110,85 @@ void TCPServer::HandleAccept(TCPConnection::pointer new_connection,
   StartAccept();
 }
 
+void TCPServer::Disconnect(TCPConnection::pointer connection)  {
+  channel_.Remove(connection);
+}
+
+
+void TCPServer::Handle(
+    const std::string &data,
+    TCPConnection::pointer connection) {
+  auto m = message::Message::Decode(data);
+  auto type = m.GetMessageType();
+  if (type == message::MessageType::kServerSetup) {
+    auto server_accept = message::ServerAccept::Decode(m.GetEncodedMessage());
+    HandleServerAccept(server_accept, connection);
+  } else if (type == message::MessageType::kRequest) {
+    auto request = message::Request::Decode(m.GetEncodedMessage());
+    HandleRequest(request, connection);
+  } else if (type == message::MessageType::kReplicate) {
+    auto replicate = message::Replicate::Decode(m.GetEncodedMessage());
+    HandleReplicate(replicate, connection);
+  } else if (type == message::MessageType::kReplicateAck) {
+    auto replicate_ack = message::ReplicateAck::Decode(m.GetEncodedMessage());
+    HandleReplicateAck(replicate_ack, connection);
+  }
```

Comment:
I think we should error out here in case we get an unrecognized message type. Also, is it possible to make this a switch statement on type?